util/src/main/java/edu/asu/diging/gilesecosystem/util/store/objectdb/DatabaseClient.java

```diff
@@ -43,10 +43,15 @@ public String generateId(String prefix, Function<String, T> f ) {
         String id = null;
         while (true) {
             id = prefix + generateUniqueId();
-            T existing = f.apply(id);
-            if (existing == null) {
+            if (f != null) {
```

Comment:
when is `f` ever null?

Reply:
https://github.com/diging/giles-eco-giles-web/blob/b27ca984caba5b9a4e6e97b3e138bfff83d08432/giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/core/service/delete/impl/DeleteDocumentService.java#L138

Since I made the change for the request to have only the document id and nepomuk in turn will delete all the files of the document I found that documents do not have any request ID associated with it like files. So I added a method to create requests with the prefix DELREQ and I don't store it since the document is already enqueued for deletion.
https://github.com/diging/giles-eco-giles-web/blob/b27ca984caba5b9a4e6e97b3e138bfff83d08432/giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/core/service/core/impl/TransactionalDocumentService.java#L89

Thus in this case I don't have a function to check for duplication.

Reply:
then the function you pass should just return null. but this change should not be necessary. The logic should be with the function to check for duplication, not here.